src/plugins/plugin.legend.js

```diff
@@ -459,58 +465,75 @@ var Legend = Element.extend({
 		}
 	},
 
+	/**
+	 * Gets the legend item, if any, at a given x,y position.
+	 * @private
+	 * @param {number} x - The x coordinate to check
+	 * @param {number} y - The y coordindate to check
+	 * @return {?Object} - The legend item found
+	 */
+	getLegendItemAt: function(x, y) {
+		var me = this;
+
+		if (x >= me.left && x <= me.right && y >= me.top && y <= me.bottom) {
+			// See if we are touching one of the dataset boxes
+			var lh = me.legendHitBoxes;
```

Comment:
All local variables should be defined on top of the method:

```javascript
_getLegendItemAt: function(x, y) {
    var me = this;
    var lh, hitBox, i, ilen;
```

Reply:
Why? This practice does not appear to be followed anywhere else in the Chart.js codebase, including the existing plugin code, from which this line is copied across exactly.

Reply:
We do follow the practice Simon mentioned in most of the codebase. There are parts that haven't been touched in awhile that don't, but we're doing it for anything new. The reason is that it results in a smaller minified file because terser is able to join the variable declarations. I don't particularly like changing the way we write our code to improve minification, so hopefully terser will one day be able to do this for us as I think that's the better approach. I've filed a request for that feature which has more details https://github.com/terser-js/terser/issues/260

Reply:
The legend plugin is quite old and its code hasn't been updated since a long time (just a few fixes sometimes). Take a look at this [more recent plugin](/chartjs/Chart.js/blob/master/src/plugins/plugin.filler.js) or the [time scale](/chartjs/Chart.js/blob/master/src/scales/scale.time.js) or these [helpers](/chartjs/Chart.js/blob/master/src/helpers/helpers.options.js) ... The idea is to pack variable definition at the same place and preferably on top of the hoisting scope. This is more consistent with how javascript handle `var` and helps the minifier to reduce the size of the code.

Reply:
Sounds reasonable. I have moved all the `var` declarations to the top.